rust/tools/authorization-logic/src/parsing/AuthLogic.g4

```diff
@@ -139,7 +145,9 @@ ATTRIBUTE: 'attribute';
 
 // Identifiers wrapped in quotes are constants whereas
 // identifiers without quotes are variables.
-ID : ('"')? [_a-zA-Z][_a-zA-Z0-9/.#:]* ('"')?;
+VARIABLE : [_a-zA-Z][_a-zA-Z0-9/.#:]*; 
+CONSTANT : '"' [_a-zA-Z+-][@_a-zA-Z0-9/.#:+=-]* '"';
+MULTILINE_CONSTANT : '"""' [\n _a-zA-Z+-][\n _a-zA-Z0-9/.#:+=-]* '"""';
```

Comment:
OK I tried refactoring it the way you were suggesting [here](https://github.com/google-research/raksha/pull/627/files#diff-f485debf22070ea87700d2b589051e492d0a3a0f9e9084f854cd242dbaad38dd) -- I do think your suggestion is better in principle.

With the way I was trying to implement your suggestion, antlr seems to handle newlines differently in some way that breaks things. When trying to parse a relation declaration, it fails to parse the relation name into the right smaller piece of syntax (here a VARIABLE, but I also tried leaving it as an ID), and this breaks all the tests. Weirdly, if I start some of the tests (like `numStringsInNames` // cargo test test_num_string_names) with a multiline comment, it suddenly parses correctly.

I spent some time trying to get it to work this way, and I could keep trying to pull on this thread if you think it is much better, but otherwise perhaps we can leave this as-is for now since we are likely to revisit the frontend of this language anyway. Let me know what you think.

Reply:
I wonder whether your specification of `CONSTANT` versus `MULTILINE_CONSTANT` is running afoul of this trap in Antlr due to them both ending with a `"`: https://github.com/antlr/antlr4/blob/master/doc/lexer-rules.md#redundant-string-literals

In that case, the `\n` could actually be disambiguating between the two because `\n` is not allowed in a regular constant.

One option could be to change a multiline constant to use three single quotes instead of double quotes. Or, perhaps, have a syntax closer to C++'s [raw string literal](https://en.cppreference.com/w/cpp/language/string_literal).